File: python/lc121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        max_profit = 0
        buy = prices[0]
        sell = prices[0]
        for i in range(1,n):            
            if prices[i] > sell:
                sell = prices[i]
                max_profit = max(max_profit, sell - buy)
            elif prices[i] < buy:
                buy = prices[i]
                sell = prices[i]
        return max_profit

File: cs/lc121.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int n = prices.Length;
        int max_profit = 0;
        int buy = prices[0];
        int sell = prices[0];
        for(int i = 1;i<n; i++)        
        {
            if (prices[i] > sell)
            {
                sell = prices[i];
                max_profit = Math.Max(max_profit, sell - buy);
            }
            else if (prices[i] < buy)
            {
                buy = prices[i];
                sell = prices[i];
            }
        }
        return max_profit;
    }
}
